src/Modules/UngradedModules.jl

```diff
@@ -3833,17 +3867,23 @@ end
 Apply $\text{Hom}(-,P)$ to `C`. Return the lifted chain complex.
 """
 function hom(C::Hecke.ChainComplex{ModuleFP}, P::ModuleFP)
-  hom_chain = Hecke.map_type(C)[]
-  hom_modules = [hom(domain(C.maps[1]),P)]
-  hom_modules = vcat(hom_modules, [hom(codomain(f), P) for f = C.maps])
+  #hom_chain = Hecke.map_type(C)[]
+  hom_chain = valtype(C.maps)[]
+  chain_range = range(C)
+  hom_modules = [hom(domain(map(C,first(chain_range))),P)]
+  hom_modules = vcat(hom_modules, [hom(codomain(map(C,i)), P) for i in chain_range])
 
-  for i=1:length(C)
+  for i=1:length(chain_range)
     A = hom_modules[i][1]
     B = hom_modules[i+1][1]
 
-    push!(hom_chain, lift_homomorphism_contravariant(B,A,map(C,i)))
+    j = chain_range[i]
+    push!(hom_chain, lift_homomorphism_contravariant(B,A,map(C,j)))
   end
-  return Hecke.ChainComplex(ModuleFP, reverse(hom_chain))
+
+  start = C.direction == :left ? C.start - length(hom_chain) : C.start
+  start = -C.start
+  return Hecke.ChainComplex(ModuleFP, reverse(hom_chain), start=start, direction=C.direction)
```

Comment:
This code is not covered by tests.

Also, it accesses members of `C` directly which should be avoided. But I think we really need to add (back?) some APIs to `ChainComplex` here to make this nicer; e.g. to avoid `C.direction` one needs to call `is_chain_complex` or `is_cochain_complex`; but then we really need `:left` or `:right` for the constructor here, and thus needs to translate this back. Also the need to compute `start` here is annoying. Not sure what the best way to deal with this is, as I am not familiar enough with `Hecke.ChainComplex`. I could make educated guesses, but perhaps @thofma or @fieker have a great suggestion off the shelf

Reply:
my guess is that `first(range(C))` or similar should do...